src/main/java/de/tum/in/www1/artemis/domain/TempIdObject.java

```diff
@@ -24,6 +24,12 @@ public void setTempID(Long tempID) {
         this.tempIDTransient = tempID;
     }
 
+    @Override
+    public int hashCode() {
+        // Important: do not include the tempId in the hash code
```

Comment:
Equality and hashcode of a `DomainObject` are defined via the id of the object. A `TempIdObject` doesn't have a permanent id or rather only a temporary id that is used to describe equality. Since the hashCode can't contain the tempId (as it's subject to change), we can't guarantee this. Or at least I understand this to be a contradiction in itself.